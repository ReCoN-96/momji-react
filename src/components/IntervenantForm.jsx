/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-boolean-value */
/* eslint-disable react/jsx-props-no-spreading */
import {
  FormControl,
  FormLabel,
  Input,
  RadioGroup,
  Radio,
  Stack,
  Box,
  Button,
} from '@chakra-ui/react';
import { useForm, Controller } from 'react-hook-form';

const IntervenantForm = ({
  profile, email, address, isActive,
}) => {
  const {
    register, handleSubmit, control,
  } = useForm({
    defaultValues: {
      'profile.firstName': profile?.firstName || '',
      'profile.lastName': profile?.lastName || '',
      email: email || '',
      address: address || '',
      isActive: isActive || false,
    },
  });

  const onSubmit = (data) => console.log(data);

  return (
    <Box w="50%" p={6} border="1px" borderColor="gray.200" borderRadius="40px">
      <form onSubmit={handleSubmit(onSubmit)}>
        <FormControl>
          <FormLabel>First name</FormLabel>
          <Input mb={2} type="text" defaultValue={profile?.firstName || ''} {...register('profile.firstName', { required: true })} />
        </FormControl>
        <FormControl>
          <FormLabel>Last name</FormLabel>
          <Input mb={2} type="text" defaultValue={profile?.lastName || ''} {...register('profile.lastName', { required: true })} />
        </FormControl>
        <FormControl>
          <FormLabel>Email address</FormLabel>
          <Input mb={2} type="email" defaultValue={email || ''} {...register('email', { required: true })} />
        </FormControl>

        <FormControl>
          <FormLabel>Home Address</FormLabel>
          <Input mb={2} type="text" defaultValue={address || ''} {...register('address', { required: true })} />
        </FormControl>

        <FormControl>
          <FormLabel>Are you active ?</FormLabel>
          <Controller
            name="isActive"
            defaultValue={isActive || false}
            control={control}
            render={({ field: { onChange, value } }) => (
              <RadioGroup mb={2} value={value} onChange={(e) => onChange(e === 'true')}>
                <Stack spacing={5} direction="row">
                  <Radio colorScheme="blue" value={true}>
                    Yes
                  </Radio>
                  <Radio colorScheme="orange" value={false}>
                    No
                  </Radio>
                </Stack>
              </RadioGroup>
            )}
          />
        </FormControl>

        <Box display="flex" justifyContent="flex-end">
          <Button type="submit" colorScheme="blue">Save</Button>
        </Box>
      </form>
    </Box>

  );
};

export default IntervenantForm;
