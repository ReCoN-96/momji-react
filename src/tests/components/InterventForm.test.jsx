/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-undef */
import renderer from 'react-test-renderer';
import IntervenantForm from '../../components/IntervenantForm';
import intervenants from '../fixtures/intervenants';

test('should render IntervenantForm correctly', () => {
  const wrapper = renderer.create(<IntervenantForm />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

test('Should render IntervenantForm correctly with intervenant data', () => {
  const wrapper = renderer.create(<IntervenantForm {...intervenants[0]} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
