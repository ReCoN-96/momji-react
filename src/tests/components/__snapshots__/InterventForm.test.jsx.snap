// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render IntervenantForm correctly with intervenant data 1`] = `
<div
  className="css-1l46jj7"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-17"
        id="field-17-label"
      >
        First name
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue="Jodie"
        disabled={false}
        id="field-17"
        name="profile.firstName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-18"
        id="field-18-label"
      >
        Last name
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue="Copeland"
        disabled={false}
        id="field-18"
        name="profile.lastName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-19"
        id="field-19-label"
      >
        Email address
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue="jodie.copeland@gmail.com"
        disabled={false}
        id="field-19"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="email"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-20"
        id="field-20-label"
      >
        Home Address
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue="930 Lewis Avenue, Bonanza, Idaho, 7682"
        disabled={false}
        id="field-20"
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-21"
        id="field-21-label"
      >
        Are you active ?
      </label>
      <div
        className="chakra-radio-group css-wlea3r"
        role="radiogroup"
      >
        <div
          className="chakra-stack css-9gj9xm"
        >
          <label
            className="chakra-radio css-78joka"
          >
            <input
              checked={true}
              className="chakra-radio__input"
              disabled={false}
              id="radio-23"
              name="radio-22"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              readOnly={false}
              required={false}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="radio"
              value={true}
            />
            <span
              aria-hidden={true}
              className="chakra-radio__control css-ssalds"
              data-checked=""
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-radio__label css-1y8kf23"
              data-checked=""
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Yes
            </span>
          </label>
          <label
            className="chakra-radio css-78joka"
          >
            <input
              checked={false}
              className="chakra-radio__input"
              disabled={false}
              id="radio-24"
              name="radio-22"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              readOnly={false}
              required={false}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="radio"
              value={false}
            />
            <span
              aria-hidden={true}
              className="chakra-radio__control css-ssalds"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-radio__label css-1y8kf23"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              No
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      className="css-1bvc4cc"
    >
      <button
        className="chakra-button css-1xx8uq7"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;

exports[`should render IntervenantForm correctly 1`] = `
<div
  className="css-1l46jj7"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-1"
        id="field-1-label"
      >
        First name
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue=""
        disabled={false}
        id="field-1"
        name="profile.firstName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-2"
        id="field-2-label"
      >
        Last name
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue=""
        disabled={false}
        id="field-2"
        name="profile.lastName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-3"
        id="field-3-label"
      >
        Email address
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue=""
        disabled={false}
        id="field-3"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="email"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-4"
        id="field-4-label"
      >
        Home Address
      </label>
      <input
        className="chakra-input css-wlea3r"
        defaultValue=""
        disabled={false}
        id="field-4"
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <div
      className="chakra-form-control css-1kxonj9"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="field-8"
        id="field-8-label"
      >
        Are you active ?
      </label>
      <div
        className="chakra-radio-group css-wlea3r"
        role="radiogroup"
      >
        <div
          className="chakra-stack css-9gj9xm"
        >
          <label
            className="chakra-radio css-78joka"
          >
            <input
              checked={false}
              className="chakra-radio__input"
              disabled={false}
              id="radio-5"
              name="radio-7"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              readOnly={false}
              required={false}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="radio"
              value={true}
            />
            <span
              aria-hidden={true}
              className="chakra-radio__control css-ssalds"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-radio__label css-1y8kf23"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Yes
            </span>
          </label>
          <label
            className="chakra-radio css-78joka"
          >
            <input
              checked={true}
              className="chakra-radio__input"
              disabled={false}
              id="radio-6"
              name="radio-7"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              readOnly={false}
              required={false}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="radio"
              value={false}
            />
            <span
              aria-hidden={true}
              className="chakra-radio__control css-ssalds"
              data-checked=""
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-radio__label css-1y8kf23"
              data-checked=""
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              No
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      className="css-1bvc4cc"
    >
      <button
        className="chakra-button css-1xx8uq7"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;
